src/main/groovy/com/bmuschko/gradle/docker/DockerRemoteApiPlugin.groovy

```diff
@@ -48,7 +48,7 @@ class DockerRemoteApiPlugin implements Plugin<Project> {
     }
 
     private void configureRegistryAwareTasks(Project project, DockerRegistryCredentials dockerRegistryCredentials) {
-        project.tasks.withType(RegistryCredentialsAware, new Action<RegistryCredentialsAware>() {
+        project.tasks.withType(RegistryCredentialsAware).configureEach( new Action<RegistryCredentialsAware>() {
```

Comment:
It's listed as one of the migration steps [here](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:task_configuration_avoidance_migration_steps).

if you search "withType(java.lang.class, org.gradle.api.action)" on that page, you can find an entry on the [old vs new API overview](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:old_vs_new_configuration_api_overview) which states

> Instead of: DomainObjectCollection.withType(java.lang.Class, org.gradle.api.Action) use withType(java.lang.Class).configureEach(org.gradle.api.Action)

Reply:
We are dealing with a `TaskCollection` here so there's no need to change it. See documentation below.

```
Instead of: TaskCollection.withType(java.lang.Class) | This is OK to use because it does not require tasks to be created immediately.
-- | --
Use: OK
```